contracts/modules/staking/StakingPool.sol

```diff
@@ -1240,18 +1248,77 @@ contract StakingPool is IStakingPool, ERC721 {
     return ownerOf(0);
   }
 
-  /* management */
+  /* pool management */
+
+  function setProducts(ProductParams[] memory params) external onlyManager {
+    uint[] memory productIds = new uint[](params.length);
+
+    for (uint i = 0; i < params.length; i++) {
+      if (params[i].recalculateEffectiveWeight) {
+        productIds[i] = params[i].productId;
+      }
+    }
+
+    (uint globalCapacityRatio, uint[] memory initialPriceRatios, uint[] memory capacityReductionRatios) = ICover(coverContract).getCapacityRatios(productIds);
+
+    uint _totalTargetWeight = totalTargetWeight;
+    uint _totalEffectiveWeight = totalEffectiveWeight;
+
+    for (uint i = 0; i < params.length; i++) {
+      ProductParams memory _param = params[i];
+      StakedProduct memory _product = products[_param.productId];
+
```

Comment:
We need to make sure that it's not possible to set the weight to a non-zero value while having the price zero.